### PR TITLE
Refactor `Package` to no longer depend on `Folder`.

### PR DESCRIPTION
**Pull request by rohinb2**

### CODE REVIEW

**Comment by rohinb2** [comment]
* **#1161** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1161?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rohinb2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/packages/package.py [hunk 1]
```diff
@@ -376,54 +383,39 @@ def to(
...
 
-        # sync folder to the fs and create a new module on the fs where relevant
-        new_folder = self.install_target.to(system, path=path)
-        new_folder = new_folder._destination_folder(
-            dest_path=new_folder.path, dest_system=system
-        )
-        new_package = copy.copy(self)
-        new_package.install_target = new_folder
-        return new_package
+            if not system._folder_exists(path_to_sync_to_on_cluster):
```
Comment (dongreenberg) [inline runhouse/resources/packages/package.py:403]: Why do we only sync if the folder doesn't already exist? How do we propogate updates?

**Review by dongreenberg — APPROVED** [review]

File: tests/conftest.py [hunk 2]
```diff
@@ -357,10 +357,10 @@ def event_loop():
 default_fixtures[TestLevels.UNIT] = {"cluster": ["named_cluster"]}
 default_fixtures[TestLevels.LOCAL] = {
     "cluster": [
-        "docker_cluster_pk_ssh_no_auth",  # Represents private dev use case
-        "docker_cluster_pk_ssh_den_auth",  # Helps isolate Auth issues
+        # "docker_cluster_pk_ssh_no_auth",  # Represents private dev use case
```
Comment (jlewitt1) [inline tests/conftest.py:360]: do these still need to be commented out?

File: runhouse/resources/packages/package.py [hunk 3]
```diff
@@ -507,11 +501,7 @@ def from_string(specifier: str, dryrun=False):
                 # Check if this is a package that was installed from local
                 local_install_path = get_local_install_path(target)
                 if local_install_path and Path(local_install_path).exists():
-                    from runhouse.resources.folders import Folder
-
-                    target = Folder(
-                        path=local_install_path, system=Folder.DEFAULT_FS, dryrun=True
-                    )
+                    target = (local_install_path, None)
```
Comment (jlewitt1) [inline runhouse/resources/packages/package.py:504]: why a tuple and not an `InstallTarget` object? (seeing we do that below also)

File: runhouse/resources/packages/package.py [hunk 4]
```diff
@@ -113,9 +126,11 @@ def _prepend_env_command(install_cmd: str, env: Union[str, "Env"] = None):
...
         # If self.path is the same as the user's home directory, raise an error.
         # Check this with Path and expanduser to handle both relative and absolute paths.
-        if Path(self.install_target.path).expanduser() in [
-            Path("~").expanduser(),
-            Path("/"),
+        if isinstance(
+            self.install_target, InstallTarget
+        ) and self.install_target.full_local_path_str() in [
+            str(Path("~").expanduser()),
+            str(Path("/")),
```
Comment (jlewitt1) [inline runhouse/resources/packages/package.py:133]: why "/" and not "~" for the cluster path?

File: runhouse/resources/packages/package.py [hunk 5]
```diff
@@ -433,14 +423,14 @@ def split_req_install_method(req_str: str):
...
     def from_config(config: dict, dryrun=False, _resolve_children=True):
-        if isinstance(config.get("install_target"), dict):
-            from runhouse.resources.folders import Folder
-
-            config["install_target"] = Folder.from_config(
-                config["install_target"],
-                dryrun=dryrun,
-                _resolve_children=_resolve_children,
-            )
+        # if isinstance(config.get("install_target"), dict):
```
Comment (jlewitt1) [inline runhouse/resources/packages/package.py:426]: can we remove now?

Comment (rohinb2) [reply runhouse/resources/packages/package.py:426]: actually... I need to mess with the whole `.config()`. good call.

File: runhouse/resources/packages/package.py (hunk 4 above)
Comment (jlewitt1) [reply runhouse/resources/packages/package.py:133]: ah ok misunderstood that to be a cluster path ref, makes sense

File: runhouse/resources/packages/package.py (hunk 5 above)
Comment (rohinb2) [reply runhouse/resources/packages/package.py]: fixed now!

**Comment by rohinb2** [comment]
### Merge activity

* **Aug 19, 1:48 PM EDT**: @rohinb2 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/1161).

**Pull request merged**